index.js

```diff
@@ -31,6 +33,7 @@ const preprocessDestinationPath = (source, destination, options) => {
 
 module.exports = (source, destination, options = {}) => {
```

Comment:
The current way of doing it means it won't fail for invalid values like `0` or `-1` as it will just fall back to the default.

This will let the `p-all` option validation work properly:

```suggestion
module.exports = (source, destination, {
	concurrency = (os.cpus().length || 1) * 2,
	...options
} = {}) => {
```

Please add a test to ensure that it correctly throws on invalid input like `-1`.